src/base-container/index.jsx

```diff
@@ -31,12 +32,19 @@ const BaseContainer = ({
 }) => {
   const dispatch = useDispatch();
 
+  const authenticatedUser = getAuthenticatedUser();
+
+  const loginErrorResult = useSelector(state => state.login.loginError);
+
   const handleOnClose = () => {
     deleteQueryParams(['authMode', 'tpa_hint', 'password_reset_token', 'track']);
     dispatch(forgotPasswordClearStatus());
     dispatch(loginErrorClear());
     dispatch(clearAllRegistrationErrors());
     close();
+    if (authenticatedUser && loginErrorResult.redirectUrl) {
```

Comment:
can we have an additional check here for currentOpenedForm = FORGOT_PASSWORD_FORM?

Reply:
i did